pkg/addons/default/aws_node_test.go

```diff
@@ -91,97 +98,98 @@ var _ = Describe("default addons - aws-node", func() {
-				Equal(preUpdateAwsNode.Spec.Template.Spec.Containers[0].Image),
-			)
-			Expect(awsNode.Spec.Template.Spec.InitContainers).To(HaveLen(1))
-			Expect(awsNode.Spec.Template.Spec.InitContainers[0].Image).To(
-				HavePrefix("961992271922.dkr.ecr.cn-northwest-1.amazonaws.com.cn/amazon-k8s-cni-init"),
-			)
+				awsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), da.AWSNode, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(awsNode.Spec.Template.Spec.Containers).To(HaveLen(1))
+				Expect(awsNode.Spec.Template.Spec.Containers[0].Image).NotTo(
+					Equal(preUpdateAwsNode.Spec.Template.Spec.Containers[0].Image),
+				)
+				Expect(awsNode.Spec.Template.Spec.InitContainers).To(HaveLen(1))
+				Expect(awsNode.Spec.Template.Spec.InitContainers[0].Image).To(
+					Equal("961992271922.dkr.ecr.cn-northwest-1.amazonaws.com.cn/amazon-k8s-cni-init:v1.9.3"),
+				)
+			})
 		})
 
-		It("detects matching image version when determining plan", func() {
-			// updating from latest to latest needs no updating
-			needsUpdate, err := UpdateAWSNode(rawClient, "eu-west-2", true)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(needsUpdate).To(BeFalse())
+		When("dry run is true", func() {
+			When("it needs an update", func() {
+				It("returns true", func() {
+					needsUpdate, err := da.UpdateAWSNode(input, true)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(needsUpdate).To(BeTrue())
+
+					awsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), da.AWSNode, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					//should be unchanged
+					Expect(awsNode.Spec).To(Equal(preUpdateAwsNode.Spec))
+				})
+			})
+
+			When("it doesn't need an update", func() {
+				BeforeEach(func() {
+					rawClient = testutils.NewFakeRawClient()
+					input.RawClient = rawClient
+					sampleAddons := testutils.LoadSamples("assets/aws-node.yaml")
+
+					rawClient.AssumeObjectsMissing = true
+
+					for _, item := range sampleAddons {
+						rc, err := rawClient.NewRawResource(item)
+						Expect(err).NotTo(HaveOccurred())
+						_, err = rc.CreateOrReplace(false)
+						Expect(err).NotTo(HaveOccurred())
+					}
+
+					var err error
+					preUpdateAwsNode, err = rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), da.AWSNode, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("returns false", func() {
+					// updating from latest to latest needs no updating
+					needsUpdate, err := da.UpdateAWSNode(input, true)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(needsUpdate).To(BeFalse())
+
+					awsNode, err := rawClient.ClientSet().AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.TODO(), da.AWSNode, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					//should be unchanged
+					Expect(awsNode.Spec).To(Equal(preUpdateAwsNode.Spec))
+				})
+			})
 		})
 	})
 })
```

Comment:
don't even attempt to review the diff, the previous test was a hot piece of garbage. Just checkout the file as a whole and see if the tests make sense plz :smile: